File: 0912-sort-an-array/0912-sort-an-array.py
class Solution:
    def MergeSort(self , arr:list[int],i : int , j : int) :
        if(i == j) :
            return
        mid = i + (j - i)//2
        self.MergeSort(arr , i , mid )
        self.MergeSort(arr,mid+1 , j)
        temp = []
        x = i
        y = mid + 1
        while (x <= mid and y <= j ) :
            if(arr[x] < arr[y]) :
                temp.append(arr[x])
                x+=1
            else :
                temp.append(arr[y])
                y+=1
        while (x <= mid) :
            temp.append(arr[x])
            x+=1
        while (y <= j) :
            temp.append(arr[y])
            y+=1
        for k in range(len(temp)) :
            arr[i+k] = temp[k]
        
    def sortArray(self, nums: list[int]) -> list[int]:
        self.MergeSort( nums , 0 , len(nums) - 1)
        return nums
    

File: 0912-sort-an-array/0912-sort-an-array.java
public class Solution {
    public void merge_sort(int[] arr , int i , int j){
        
        int mid = i +(j-i)/2;
        if(i == j){
            return ;
        }
         merge_sort(arr,i,mid);
         merge_sort(arr,mid+1,j);
        int x = i;
        int y = mid+1;
        int n = mid - i + 1;
        int m = j - mid + 1;
       int[] temp = new int[j-i+1];
        int k = 0;
        while(x <= mid && y<= j){
            if(arr[x] < arr[y] ){
            temp[k++] = arr[x++];    
            }
            else{
                temp[k++] = arr[y++];
            }       

        }

        while(x<= mid)
        temp[k++] = arr[x++];
        

        while(y<=j)
        temp[k++] = arr[y++];

        for(int p = 0 ; p< (j-i+1) ; p++){
            arr[i+p] = temp[p];
        }

        
    }
    public int[] sortArray(int[] nums) {
        int i = 0;
        int j = nums.length;
         merge_sort(nums,i,j-1);
         return nums;

        
    }
}